### PR TITLE
Infix search

### PR DESCRIPTION
**Pull request by eavgerinos**

### CODE REVIEW

**Comment by Zorbash** [comment]
+1 LGTM

**Comment by mkousta** [comment]
+1

**Comment by alup** [comment]
I would prefer to use an option to be able to select 'prefix' or 'infix' search.

**Comment by eavgerinos** [comment]
@alup functionallity added. By default Selectorablium will use infix.

File: coffee/storage_freak.coffee [hunk 1]
```diff
@@ -177,15 +180,15 @@ define [
...
+    _createAccentIndependentRE: (query, type="infix")->
       for value in @config.list_of_replacable_chars
         re = new RegExp "#{value[0]}|#{value[1]}", 'ig'
         query = query.replace re, "(?:#{value[0]}|#{value[1]})"
 
       re = null
-      return new RegExp "^#{query}", 'ig'
-
 
+      return new RegExp "#{query}", 'ig' if type == "infix"
```
Comment (alup) [inline coffee/storage_freak.coffee]: Prefer using `if/else` instead of this ... E.g.

**Comment by alup** [comment]
+1

File: coffee/storage_freak.coffee (hunk 1 above)
Comment (eavgerinos) [inline coffee/storage_freak.coffee]: meh

File: coffee/storage_freak.coffee [hunk 2]
```diff
@@ -177,15 +180,15 @@ define [
       @_set @_timestamp_key, new Date().getTime()
       @_trigger 'dbupdated'
 
-    _createAccentIndependentRE: (query)->
+    _createAccentIndependentRE: (query, type="infix")->
```
Comment (Zorbash) [inline coffee/storage_freak.coffee]: Add spaces around `=`

File: spec/storagefreak_spec.coffee [hunk 3]
```diff
@@ -425,6 +425,42 @@ describe 'StorageFreak', ->
...
+
+        @instance.on 'dbsearch_results', (data, query, remote = false)->
+          expect(data)
+            .to.be.an('array')
+            .to.have.length(2)
+          done()
+
+        @instance.search('as')
+
+
```
Comment (Zorbash) [inline spec/storagefreak_spec.coffee]: :scissors:

File: spec/storagefreak_spec.coffee [hunk 4]
```diff
@@ -425,6 +425,42 @@ describe 'StorageFreak', ->
...
       expect(spy).to.not.be.called
 
+    it 'matches only infix data by default', (done)->
+      @instance._data = {
+        add: 'addas'
+        asd: 'asd'
+        as: 'as'
+      }
+
+      @instance.on 'dbsearch_results', (data, query, remote = false)->
```
Comment (Zorbash) [inline spec/storagefreak_spec.coffee]: `query`, `remote` don't seem to be used in this anon function, you may remove them

File: spec/storagefreak_spec.coffee [hunk 5]
```diff
@@ -425,6 +425,42 @@ describe 'StorageFreak', ->
       @instance.search('as')
       expect(spy).to.not.be.called
 
+    it 'matches only infix data by default', (done)->
+      @instance._data = {
+        add: 'addas'
```
Comment (Zorbash) [inline spec/storagefreak_spec.coffee]: You could change `add: 'addas`' to `addasadd: 'addasadd'` to make it even clearer that we're operating in `infix` mode. (We could keep `addas` for `suffix` mode).

**Comment by Zorbash** [comment]
+1 LGTM

File: spec/storagefreak_spec.coffee (hunk 5 above)
Comment (eavgerinos) [inline spec/storagefreak_spec.coffee]: There will be no `suffix` mode. There is no use case

Comment (eavgerinos) [inline spec/storagefreak_spec.coffee]: Still I'm changing it

File: coffee/storage_freak.coffee [hunk 6]
```diff
@@ -177,15 +180,15 @@ define [
       @_set @_timestamp_key, new Date().getTime()
       @_trigger 'dbupdated'
 
-    _createAccentIndependentRE: (query)->
+    _createAccentIndependentRE: (query, type = "infix")->
```
Comment (cvkef) [inline coffee/storage_freak.coffee]: minor: single quotes

**Pull request merged**